### PR TITLE
Fix: Properly handle paths with spaces in generated command

### LINKED ISSUE #666: air doesn't read the spaces between names of directories
#https://github.com/air-verse/air/issues/630#event-14693459937
check above thread for more info

### PR DESCRIPTION
**Pull request by pkashin**
- Fixes incorrect command generation when the binary path contains spaces (issue #666).
- Wraps paths with spaces in quotes to prevent shell splitting.
- Adds OS-specific handling for Windows (uses `& "..."` syntax).

**Changes:**
1. Added `formatPath()` function in `runner/util.go` to:
   - Quote absolute paths with spaces.
   - Apply Windows-specific syntax when needed.
2. Integrated `formatPath()` into command generation in `runner/engine.go`.
3. Added comprehensive tests in `runner/util_test.go` covering:
   - Windows/Unix absolute/relative paths with/without spaces.
   - Edge cases (empty path, simple filename).

**Testing:**
- Unit tests verify correct behavior across platforms.
- Manually tested on [Windows 11/macOS 15] with paths containing spaces.

**Notes:**
- This ensures commands like `C:\Program Files\app.exe` or `/usr/local/my app/main` execute correctly.

Closes #666

### CODE REVIEW

**Comment by pkashin** [comment]
Hi @xiantang,

I noticed that the `Smoke test / smoke_test_ubuntu / Smoke test (pull_request)` check failed ❌ in this PR. However, the same test passes successfully ✅ in my fork (screenshot attached below). Could you please help me investigate this issue?

It would be great if you could:

Restart the failed check to see if it was a transient issue.

Let me know if there are any additional steps I should take to resolve this.

Thank you for your time and assistance!

Снимок экрана 2025-03-03 в 22 44 48

**Comment by xiantang** [comment]
looks pipeline is ok now.

File: runner/util.go [hunk 1]
```diff
@@ -429,3 +429,17 @@ func joinPath(root, path string) string {
...
+
+func formatPath(path string) string {
+	if !filepath.IsAbs(path) || !strings.Contains(path, " ") {
+		return path
+	}
+
+	quotedPath := fmt.Sprintf(`"%s"`, path)
+
+	if runtime.GOOS == PlatformWindows {
+		return fmt.Sprintf(`& %s`, quotedPath)
```
Comment (xiantang) [inline runner/util.go:441]: im not using Windows, what does `&` mean?

Comment (pkashin) [reply runner/util.go:441]: Adding `&` before the path in quotation marks ("path") is necessary for PowerShell or the Windows command prompt to correctly interpret the path with spaces as a command to execute. Without `&`, a path with spaces can be interpreted as a string, not as a command.
Usage examples:
1. Running an executable file with spaces in the path:
If the path to the executable file contains spaces, PowerShell will not be able to execute it without `&`. 
```powershell
& "C:\Program Files\MyApp\myapp.exe"
```
2. Running commands with arguments:
You can use & to run commands with arguments:
```powershell
& "C:\Program Files\MyApp\myapp.exe" --arg1 value1
```

Comment (pkashin) [reply runner/util.go:441]: Notes:
- I ran the tests manually locally on Windows, then the coverage reaches 100% for the `formatPath()` function.

**Review by xiantang — APPROVED** [review]
LGTM
